ui/src/pages/topics/[id]/index.tsx

```diff
@@ -36,21 +44,73 @@ export const getServerSideProps: Types.GetServerSideProps = async (context) => {
 
     return {
         props: {
-            topic
+            topic,
+            bookmarkId
         }
     };
 };
 
 type Props = {
     topic: Interfaces.Topic;
+    bookmarkId: string | null;
 };
 
 const Topic: Types.NextPage<Props> = (props): React.ReactElement => {
     const user = Stores.useAuthStore((state) => state.user);
 
+    const isBookmarked = props.bookmarkId ? true : false;
+    const [showBookmark, setShowBookmark] = React.useState(isBookmarked);
+    const [bookmarkId, setBookmarkId] = React.useState(props.bookmarkId);
+
     const topic = props.topic;
     const showChildren = Boolean(topic.children.length);
     const showParents = Boolean(topic.parents.length);
+
+    useEffect(() => {
+        setShowBookmark(isBookmarked);
```

Comment:
On this page we have the same bug regarding bookmarking mechanism. The same fix what i suggested for publications can be applied here also.